rspace/src/test/scala/coop/rchain/rspace/ReplayRSpaceTests.scala

```diff
@@ -39,45 +47,52 @@ trait ReplayRSpaceTests
       persist: Boolean
   )(
       implicit matcher: Match[P, Nothing, A, R]
-  ): ParSeq[Option[(ContResult[C, P, K], Seq[Result[R]])]] =
-    (if (shuffle) Random.shuffle(range.toList) else range.toList).par.map { i: Int =>
+  ): Task[List[Option[(ContResult[C, P, K], Seq[Result[R]])]]] =
+    (if (shuffle) Random.shuffle(range.toList) else range.toList).parTraverse { i: Int =>
       logger.debug("Started consume {}", i)
-      val res =
-        space.consume(channelsCreator(i), patterns, continuationCreator(i), persist).right.get
-      logger.debug("Finished consume {}", i)
-      res
+      space
+        .consume(channelsCreator(i), patterns, continuationCreator(i), persist)
+        .map { r =>
+          val unpacked = r.right.get
+          logger.debug("Finished consume {}", i)
+          unpacked
+        }
     }
 
   def produceMany[C, P, A, R, K](
-      space: IdISpace[C, P, Nothing, A, R, K],
+      space: ISpace[Task, C, P, Nothing, A, R, K],
       range: Range,
       shuffle: Boolean,
       channelCreator: Int => C,
       datumCreator: Int => A,
       persist: Boolean
   )(
       implicit matcher: Match[P, Nothing, A, R]
-  ): ParSeq[Option[(ContResult[C, P, K], immutable.Seq[Result[R]])]] =
-    (if (shuffle) Random.shuffle(range.toList) else range.toList).par.map { i: Int =>
+  ): Task[List[Option[(ContResult[C, P, K], immutable.Seq[Result[R]])]]] =
+    (if (shuffle) Random.shuffle(range.toList) else range.toList).parTraverse { i: Int =>
       logger.debug("Started produce {}", i)
-      val res = space.produce(channelCreator(i), datumCreator(i), persist).right.get
-      logger.debug("Finished produce {}", i)
-      res
+      space.produce(channelCreator(i), datumCreator(i), persist).map { r =>
+        val unpacked = r.right.get
+        logger.debug("Finished produce {}", i)
+        unpacked
+      }
     }
 
   "reset to a checkpoint from a different branch" should "work" in withTestSpaces {
     (space, replaySpace) =>
-      val root0 = replaySpace.createCheckpoint().root
-      replaySpace.store.isEmpty shouldBe true
+      (for {
+        root0 <- replaySpace.createCheckpoint().map(_.root)
+        _     = replaySpace.store.isEmpty shouldBe true
```

Comment:
Does delaying this make sense?

Reply:
we know that task will behave as expected so... no, I don't think there is reason to do so.